### ISSUE TITLE
axe does not recognize input label when supplied by role="img" with aria-labelledby

### ISSUE THREAD

**Issue by gerardkcohen** [dequelabs/axe-core#4389]
### Product

axe Extension

### Product Version

Extension: v4.77.1 axe-core: v4.8.4

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
axe should recognize an ARIA image using `aria-describedby` to supply alternate text as a label for input fields.

#### Actual
axe does not recognize an input label when it is supplied using an ARIA image with `aria-describedby` for the alternate text.
axe-core returns the error that "Form elements must have a label" https://dequeuniversity.com/rules/axe/4.8/label?application=AxeChrome

#### How to Reproduce
I know this is a horrible example, but it should work based on name calculation rules.
```
<body>
  <input id="value-unset" />
  <label for="value-unset">
    <span aria-labelledby="image-label" role="img">
      <span id="image-label">unset</span>
    </span>
  </label>
</body>
```

#### Additional context
Quickly tested in various ways to make sure that the name was being calculated propery:

- Chrome dev tools accessibility panels reports the name for the input as "unset"
- Test with Chrome and VoiceOver and the input field was announced as "unset"
- WAVE plugin does not report this as an error

**Comment by straker** [comment]
Thanks for the issue. This is expected as Firefox has issues with this type of setup. [Firefox stops looking at ID references after encountering a `for` attribute](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/aria/arialabelledby-text.js#L24-L26) and so will not read the `aria-labelledby` of the `role=img` element. 

I retested Firefox and all our other browsers and assistive technologies and these are the results (note Firefox still has issues with the pattern).

* VO / Safari
  * Tabbing to the input announces "unset, edit text"
  * The rotor menu lists the input as "unset, edit text"
* NVDA / Firefox
  * Tabbing to the input announces "edit has autocomplete, blank" (no  name was read)
* JAWS / Chrome
  * Tabbing to the input announces "unset, blank"
